lib/actions.ts

```diff
@@ -286,17 +287,25 @@ export async function getJobObject(octokit: InstanceType<typeof GitHub>): Promis
     throw error
   })
 
-  const currentJob = workflowRunJobs.find((job) => job.name === absoluteJobName)
-  if (!currentJob) {
+  
+  const runnerName = getInput('runner-name', {required: true})
+  //In the case of truncated job name the only other shared identifier is the runner name
+  const currentJob = workflowRunJobs.filter((job) => job.name === absoluteJobName && job.status=== "in_progress" && job.runner_name === runnerName)
+  if (currentJob.length === 0) {
     throw new Error(`Current job '${absoluteJobName}' could not be found in workflow run.\n` +
         'If this action is used within a reusable workflow, ensure that ' +
         'action input \'workflow-context\' is set to ${{ inputs.workflow-context }}' +
         'and workflow input \'workflow-context\' was set to \'"CALLER_JOB_NAME", ${{ toJSON(matrix) }}\'' +
         'or \'"CALLER_JOB_NAME", ${{ toJSON(matrix) }}, ${{ inputs.workflow-context }}\' in case of a nested workflow.'
     )
+  } else if (currentJob.length != 1) {
+    throw new Error(`Current job '${absoluteJobName}' returned multiple matches'.\n` +
+      'If this action is used within a reusable workflow, or matrix please ensure that the job name is unique.' +
+      'If the length of \'"CALLER_JOB_NAME" + ${{ toJSON(matrix) }}\' exceeds 97 characters' +
```

Comment:
should be 100 characters as well, right?

Reply:
Yes. good catch.